apps/mochi-web/components/app/detail/AppDetailIntegration.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  Button,
+  TextFieldInput,
+  TextFieldRoot,
+  Tooltip,
+  Typography,
+} from '@consolelabs/core'
+import { CopySolid, ThreeDotLoading } from '@consolelabs/icons'
+import { truncate } from '@dwarvesf/react-utils'
+import { useClipboard } from '@dwarvesf/react-hooks'
+import { CodeSnippet } from './CodeSnippet'
+
+interface Props {
+  apiKey?: string
+}
+
+export const AppDetailIntegration = ({ apiKey = '' }: Props) => {
+  const { onCopy, hasCopied } = useClipboard(apiKey)
+
+  return (
+    <div className="mt-8">
+      <div className="py-2">
+        <Typography level="p4" className="font-medium">
+          Integrations
+        </Typography>
+      </div>
+      <div className="grid grid-cols-1 gap-8 p-6 mt-4 border sm:grid-cols-2 rounded-xl border-neutral-200 bg-neutral-0">
+        <div>
+          <Typography level="p3" className="font-medium">
+            Test API Key
+          </Typography>
+          <Typography level="p5" color="textSecondary" className="py-3">
+            Make a sample request to any Mochi product with the key below.
+          </Typography>
+          <div className="rounded border border-neutral-300 shadow-input px-4 py-1.5 flex items-center gap-2 w-fit">
+            {apiKey ? (
+              <>
+                <Typography level="h8">{truncate(apiKey, 12, true)}</Typography>
+                <Tooltip
+                  content={hasCopied ? 'Copied' : 'Click to copy API key'}
+                  arrow="top-center"
+                  componentProps={{ root: { open: hasCopied || undefined } }}
+                >
+                  <CopySolid
```

Comment:
Please use the correct icon. Also, this should be a button. Can you use IconButton here?
<img width="228" alt="image" src="https://github.com/consolelabs/web-foundation/assets/12707960/9876d3d0-2246-4d5f-8f46-cc8e8fdf316b">